txcache/txByHashMap.go

```diff
@@ -77,6 +104,20 @@ func (txMap *txByHashMap) RemoveTxsBulk(txHashes [][]byte) uint32 {
 	return numRemoved
 }
 
+// RemoveTxsBulkWithCheck removes transactions, in bulk, but checks if each tx can be removed
+func (txMap *txByHashMap) RemoveTxsBulkWithCheck(txHashes [][]byte, txTracker *transactionsTracker) uint32 {
```

Comment:
this requires a nil check on txTracker, otherwise it can panic

also, *transactionsTracker can be replaced with an interface